pkgs/racket-benchmarks/tests/racket/benchmarks/shootout/spectralnorm-par.rkt

```diff
@@ -18,44 +15,46 @@
     (for-each touch fs)))
 
 
-;; the big let improves performance by about 20%
-(let* ()
-  (define N (command-line #:args (n) (string->number n)))
-  (define C (processor-count))
-  (define (A i j)
-    (let ([ij (fx+ i j)])
-      (fl/ 1.0 (fl+ (fl* (fl* (fx->fl ij)
-                              (fx->fl (fx+ ij 1)))
-                         0.5) 
-                    (fx->fl (fx+ i 1))))))
-  (define (Av x y N)
-    (for/par C ([i N])
-      (flvector-set!
-       y i
-       (let L ([a 0.0] [j 0])
-         (if (fx= j N) a
-             (L (fl+ a (fl* (flvector-ref x j) (A i j)))
-                (fx+ j 1)))))))
-  (define (Atv x y N)
-    (for/par C ([i N])
-      (flvector-set!
-       y i
-       (let L ([a 0.0] [j 0])
-         (if (fx= j N) a
-             (L (fl+ a (fl* (flvector-ref x j) (A j i)))
-                (fx+ j 1)))))))
-  (define (AtAv x y t N) (Av x t N) (Atv t y N))
-  (define u (make-flvector N 1.0))
-  (define v (make-flvector N))
-  (define t (make-flvector N))
-  (for ([i (in-range 10)])
-    (AtAv u v t N) (AtAv v u t N))
-  (displayln (real->decimal-string 
-              (flsqrt 
-               (let L ([vBv 0.0] [vv 0.0] [i 0])
-                 (if (fx= i N) (fl/ vBv vv)
-                     (let ([ui (flvector-ref u i)] [vi (flvector-ref v i)])
-                       (L (fl+ vBv (fl* ui vi))
-                          (fl+ vv (fl* vi vi))
-                          (fx+ i 1))))))
-              9)))
+(define N (command-line #:args (n) (string->number n)))
+(define C (processor-count))
+(displayln C)
+(define (A i j)
+  (let ([ij (fx+ i j)])
+    (fl/ 1.0 (fl+ (fl* (fl* (fx->fl ij)
+                            (fx->fl (fx+ ij 1)))
+                       0.5) 
+                  (fx->fl (fx+ i 1))))))
+(define (Av x y N)
+  (for/par C ([i N])
+           (flvector-set!
+            y i
+            (let L ([a 0.0] [j 0])
+              (if (fx= j N) a
+                  (L (fl+ a (fl* (flvector-ref x j) (A i j)))
+                     (fx+ j 1)))))))
+(define (Atv x y N)
+  (for/par C ([i N])
+           (flvector-set!
+            y i
+            (let L ([a 0.0] [j 0])
+              (if (fx= j N) a
+                  (L (fl+ a (fl* (flvector-ref x j) (A j i)))
+                     (fx+ j 1)))))))
+(define (AtAv x y t N) (Av x t N) (Atv t y N))
+(define u (make-flvector N 1.0))
+(define v (make-flvector N))
+(define t (make-flvector N))
+(visualize-futures
```

Comment:
Along the same lines, was this use of `visualize-futures` meant to be part of the commit?